pennylane_lightning_gpu/src/simulator/MPIWorker.hpp

```diff
@@ -247,26 +265,22 @@ make_shared_mpi_worker(custatevecHandle_t handle, MPIManager &mpi_manager,
     size_t transferWorkspaceSize; // In bytes and its value should be power
                                   // of 2.
 
-    if (log2_mpi_buf_counts == 0) {
+    if (mpi_buf_size == 0) {
         transferWorkspaceSize = size_t{1} << numLocalQubits;
         if constexpr (std::is_same_v<CFP_t, cuDoubleComplex> ||
                       std::is_same_v<CFP_t, double2>) {
             transferWorkspaceSize = transferWorkspaceSize * sizeof(double) * 2;
         } else {
             transferWorkspaceSize = transferWorkspaceSize * sizeof(float) * 2;
         }
-        // Here 26 is based on the benchmark tests on the Perlmutter.
+        // 26 her is based on the benchmark tests on the Perlmutter and
+        // transferWorkspaceSize is 64 MB.
         if (transferWorkspaceSize > (size_t{1} << 26)) {
             transferWorkspaceSize = size_t{1} << 26;
         }
     } else {
-        transferWorkspaceSize = size_t{1} << log2_mpi_buf_counts;
-        if constexpr (std::is_same_v<CFP_t, cuDoubleComplex> ||
-                      std::is_same_v<CFP_t, double2>) {
-            transferWorkspaceSize = transferWorkspaceSize * sizeof(double) * 2;
-        } else {
-            transferWorkspaceSize = transferWorkspaceSize * sizeof(float) * 2;
-        }
+        transferWorkspaceSize =
+            size_t{1} << (mpi_buf_size + 20); // size_t{1} << 20 is 1MB
```

Comment:
I think we should define and gather computational constants somewhere. Maybe a constant namespace at the top of the module?

Reply:
Thanks Vincent! I would like to add a inline function to convert mpi_buf_size which in "MB" to "Bytes".